capacity4more/modules/c4m/features_og/c4m_features_og_manage/c4m_manage_categories.inc

```diff
@@ -0,0 +1,834 @@
+  }
+
+  $title['element'] = array(
+    '#tag' => 'h2',
+    '#attributes' => array(
+      'class' => array('term-edit-title-text'),
+    ),
+    '#value' => t('Manage Group categories'),
+  );
+
+  $description['element'] = array(
+    '#tag' => 'h4',
+    '#attributes' => array(
+      'class' => array('term-edit-description-text'),
+    ),
+    '#value' => t('Change the category term of this Group.'),
+  );
+
+  $form['title'] = array(
+    '#markup' => theme_html_tag($title),
+    '#weight' => -10,
+  );
+
+  $form['description'] = array(
+    '#markup' => theme_html_tag($description),
+    '#weight' => -5,
+  );
+
+  $form['parent'] = array(
+    '#type' => 'radios',
+    '#title' => t('Type of category'),
+    '#options' => $options,
+    '#default_value' => reset($parent),
+    '#required' => TRUE,
+    '#weight' => 0,
+  );
+
+  $form['name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Term name'),
+    '#default_value' => $term->name,
+    '#maxlength' => 255,
+    '#required' => TRUE,
+    '#weight' => 5,
+  );
+
+  $form['vocabulary_machine_name'] = array(
+    '#type' => 'value',
+    '#value' => isset($term->vocabulary_machine_name) ? $term->vocabulary_machine_name : $vocabulary->name,
+  );
+
+  $form['vid'] = array(
+    '#type' => 'value',
+    '#value' => $vocabulary->vid,
+  );
+  $form['tid'] = array(
+    '#type' => 'value',
+    '#value' => $term->tid,
+  );
+
+  $form['actions'] = array('#type' => 'actions');
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save'),
+    '#weight' => 10,
+  );
+
+  $form['actions']['cancel'] = array(
+    '#type' => 'submit',
+    '#value' => t('Cancel'),
+    '#weight' => 15,
+  );
+
+  $form['actions']['delete'] = array(
+    '#type' => 'submit',
+    '#value' => t('Delete'),
+    '#weight' => 20,
+  );
+
+  return $form;
+}
+
+/**
+ * Validation handler for the custom term form.
+ *
+ * @see c4m_features_og_manage_categories_term_form()
+ */
+function c4m_features_og_manage_categories_term_form_validate($form, &$form_state) {
+  entity_form_field_validate('taxonomy_term', $form, $form_state);
+}
+
+/**
+ * Submit handler to update a term.
+ *
+ * @see c4m_features_og_manage_categories_term_form()
+ */
+function c4m_features_og_manage_categories_term_form_submit($form, &$form_state) {
+  if ($form_state['triggering_element']['#value'] == t('Delete')) {
+    // Execute the term deletion.
+    if ($form_state['values']['delete'] === TRUE) {
+      $group = c4m_og_current_group();
+      $form['#vocabulary'] = c4m_og_vocab_load_group_vocabulary($group, 'c4m_vocab_category');
+      c4m_features_og_manage_taxonomy_term_confirm_delete_submit($form, $form_state);
+      return;
+    }
+    // Rebuild the form to confirm term deletion.
+    $form_state['rebuild'] = TRUE;
+    $form_state['confirm_delete'] = TRUE;
+    return;
+  }
+  if ($form_state['triggering_element']['#value'] == t('Cancel')) {
```

Comment:
No need for this if you use the [confirm_form](https://api.drupal.org/api/drupal/modules%21system%21system.module/function/confirm_form/7).

Parameters

* $form: Additional elements to add to the form. These can be regular form elements, #value elements, etc., and their values will be available to the submit handler.
* $question: The question to ask the user (e.g. "Are you sure you want to delete the block <em>foo</em>?"). The page title will be set to this value.
* **$path: The page to go to if the user cancels the action**. This can be either:
  * A string containing a Drupal path.
  * An associative array with a 'path' key. Additional array values are passed as the $options parameter to l().
  * If the 'destination' query parameter is set in the URL when viewing a confirmation form, that value will be used instead of $path.
* $description: Additional text to display. Defaults to t('This action cannot be undone.').
* $yes: A caption for the button that confirms the action (e.g. "Delete", "Replace", ...). Defaults to t('Confirm').
* $no: A caption for the link which cancels the action (e.g. "Cancel"). Defaults to t('Cancel').
* $name: The internal name used to refer to the confirmation item.